src/builtins.ts

```diff
@@ -113,11 +114,24 @@ function any(x: collections.Iterable): boolean {
 }
 
 function bool(x) {
-    if (x === undefined)
+    
+    if(x instanceof Py_Tuple || x instanceof iterator.XRange || x instanceof pytypes.Py_Str 
+       || x instanceof Py_List || x instanceof Py_Dict){
+        return x.len() !== 0;
+    }
+    if(x instanceof Py_Int || x instanceof Py_Float){
+        return x.toNumber() != 0;
+    }
+    if(x instanceof Py_Long){
+        return !x.eq(Py_Long.fromInt(0));
+    }
+    if(x instanceof Py_Complex){
+        return !(x.real.value == 0 && x.imag.value == 0);
+    }
+    if(x == singletons.None || x == false){
         return false;
-    if (x['bool'] !== undefined)
-        return x.bool();
-    return x != 0;
+    }
+    return true;
```

Comment:
I found the former to be more readable. There are only a few cases in which `bool()` returns false, by explicitly returning true it's clear that if none of the rules are satisfied, it's true.